### PR TITLE
feat: introduced root level sentat for batch request

### PR DESCRIPTION
**Pull request by MoumitaM**
## PR Description

Please include a summary of the change along with the relevant motivation and context.

## Linear task (optional)

Linear task link

## Cross Browser Tests

Please confirm you have tested for the following browsers:

- [ ] Chrome
- [ ] Firefox
- [ ] IE11

## Sanity Suite

- [ ] All sanity suite test cases pass locally

## Security

- [ ] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

File: packages/analytics-v1.1/src/utils/beaconQueue.js [hunk 1]
```diff
@@ -58,7 +58,7 @@ class BeaconQueue {
     queue = queue.slice(-(this.maxItems - 1));
     queue.push(message);
     let batch = queue.slice(0);
-    const data = { batch };
+    const data = { batch, sentAt: new Date().toISOString() };
```
Comment (bardisg) [inline packages/analytics-v1.1/src/utils/beaconQueue.js:61]: you ca use teh same import { getCurrentTimeFormatted } from '@rudderstack/analytics-js-common/utilities/timestamp'; here too or is there any blocker?

File: packages/analytics-v1.1/src/utils/beaconQueue.js [hunk 2]
```diff
@@ -90,10 +90,11 @@ class BeaconQueue {
   flushQueue(batch) {
     // check batch payload has data before flushing
     if (batch && batch.length > 0) {
+      const currentTime = new Date().toISOString();
```
Comment (bardisg) [inline packages/analytics-v1.1/src/utils/beaconQueue.js:93]: possible to use import { getCurrentTimeFormatted } from '@rudderstack/analytics-js-common/utilities/timestamp'; here?

File: packages/analytics-js-plugins/src/xhrQueue/utilities.ts [hunk 3]
```diff
@@ -6,13 +6,14 @@ import type { ApplicationState } from '@rudderstack/analytics-js-common/types/Ap
...
 import { clone } from 'ramda';
+import { getCurrentTimeFormatted } from '@rudderstack/analytics-js-common/utilities/timestamp';
 import { checks, http, url, json, eventsDelivery } from '../shared-chunks/common';
 import { DATA_PLANE_API_VERSION, DEFAULT_RETRY_QUEUE_OPTIONS, XHR_QUEUE_PLUGIN } from './constants';
 import type { XHRRetryQueueItemData, XHRQueueItemData, XHRBatchPayload } from './types';
 import { EVENT_DELIVERY_FAILURE_ERROR_PREFIX } from './logMessages';
 
 const getBatchDeliveryPayload = (events: RudderEvent[], logger?: ILogger): Nullable<string> => {
-  const batchPayload: XHRBatchPayload = { batch: events };
+  const batchPayload: XHRBatchPayload = { batch: events, sentAt: getCurrentTimeFormatted() };
```
Comment (bardisg) [inline packages/analytics-js-plugins/src/xhrQueue/utilities.ts:16]: does getBatchDeliveryPayload get caled on every retry?

Comment (MoumitaM) [reply packages/analytics-js-plugins/src/xhrQueue/utilities.ts:16]: Yes

File: packages/analytics-v1.1/src/utils/beaconQueue.js (hunk 1 above)
Comment (MoumitaM) [reply packages/analytics-v1.1/src/utils/beaconQueue.js]: Addressed

File: packages/analytics-v1.1/src/utils/beaconQueue.js (hunk 2 above)
Comment (MoumitaM) [reply packages/analytics-v1.1/src/utils/beaconQueue.js]: Addressed

**Review by saikumarrs — CHANGES_REQUESTED** [review]

File: packages/analytics-js-plugins/src/xhrQueue/utilities.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (saikumarrs) [reply packages/analytics-js-plugins/src/xhrQueue/utilities.ts:16]: Gotcha. Let's update the implementation to pass the timestamp to that final event mutator function.
In the batch payload preparer function, we can use the timestamp from the first event. Or, you can also pass that same timestamp to this function.

BTW, we'll also fix another bug with this change.

You can use your discretion to decide on the best approach.

[the remark this answers is not in the document]
Comment (MoumitaM) [reply packages/analytics-js-plugins/src/xhrQueue/utilities.ts]: Addressed

File: packages/analytics-js-plugins/src/beaconQueue/utilities.ts [hunk 4]
```diff
@@ -23,6 +24,7 @@ import type { BeaconBatchData } from './types';
 const getBatchDeliveryPayload = (events: RudderEvent[], logger?: ILogger): Blob | undefined => {
   const data: BeaconBatchData = {
     batch: events,
+    sentAt: getCurrentTimeFormatted(),
```
[the remark this answers is not in the document]
Comment (MoumitaM) [reply packages/analytics-js-plugins/src/beaconQueue/utilities.ts]: Addressed

**Review by saikumarrs — APPROVED** [review]

**Review by bardisg — APPROVED** [review]

**Pull request merged**